### ISSUE TITLE
Production Key not working

### ISSUE THREAD

**Issue by Sam0711er** [xing/XNGAPIClient#66]
Hey, Using the test key wasn't a problem, but now the production key won't work. How to configure that in the project? The problem I get is "CALLBACK_URL_NOT_ALLOWED
The oauth_callback URL isn't valid. The callback domain for this consumer is 'eckert.sam.url://xingapp<Consumer key>' or 'eckert.sam.url://<consumerkey>'

**Comment by pietbrauer** [comment]
So is this about the Key or the Callback URL? From your description I think it is about the Callback URL.

Did you change the Callback URL between Test and production?

**Comment by Sam0711er** [comment]
No

**Comment by pietbrauer** [comment]
Can you please provide more information on that? Like code samples and screenshots? Where is the error thrown? Is it from an HTTP request or in your app?

Not having information makes it very hard to debug it.

**Comment by Sam0711er** [comment]
Ok,
I have test and production Keys

Test Consumer Key: <other characters>e9ce
Test Consumer Secret: <other characters>0499
Product Consumer Key: <other characters>dd89
Product Consumer Secret: <other characters>1fe2

On the XING dev portal, I set up two callbackURLs:

1:    eckert.sam.url://xingapp<other characters>dd89
2:    eckert.sam.url://<other characters>dd89

In my projects info.plist i set this up for the URL identifier:
eckert.sam.url://

And this for the URL scheme:
xingapp<othercharacters>dd89

Then, in my project in the class where it's requesting authetification I have this:

 XNGAPIClient *client = [XNGAPIClient sharedClient];
        client.consumerKey = @"<other characters>dd89";
        client.consumerSecret = @"<other characters>1fe2";


When trying to call that IBAction I get the following UIAlertView (AlertView for failure):

                          failure:^(NSError *error) {
                              UIAlertView *alert = [[UIAlertView alloc] initWithTitle:@"Oops" message:@"The Xing Authentication failed. Make sure you use the right credentials." delegate:nil cancelButtonTitle:@"Dismiss" otherButtonTitles: nil];
                              [alert show];
                          }];

In the XING Dev portal, I then get the following issue:

![bildschirmfoto 2015-06-04 um 16 55 54](https://cloud.githubusercontent.com/assets/7760416/7986923/ca6bf378-0ada-11e5-81fc-205269b21cde.png)

**Comment by Sam0711er** [comment]
You have an idea, @pietbrauer ?

**Comment by pietbrauer** [comment]
To debug it I would:
* go to Notes.app on your phone
* type in `eckert.sam.url://`
* hit save
* Tap the link

Does it go into your app?

**Comment by Sam0711er** [comment]
No, it didn't. What I know did, was setting everything to this:
![bildschirmfoto 2015-06-05 um 10 56 41](https://cloud.githubusercontent.com/assets/7760416/8002762/b017537c-0b71-11e5-96aa-822047343cb0.png)

**Comment by pietbrauer** [comment]
I did a small test project. I always set it in Target -> Info -> URL Types. It looks like this:
![screen shot 2015-06-05 at 11 03 38](https://cloud.githubusercontent.com/assets/759730/8002895/b3da6dfe-0b72-11e5-9a8e-bf4a31cf7df4.png)

This registers `eckert.sam.url://`

**Comment by pietbrauer** [comment]
Or you could set your Callback URL in `dev.xing.com` to `cardbuddy://` that should do the same trick.

**Comment by Sam0711er** [comment]
Ok, thanks. But the problem is, on the dev homepage, you can't just register cardbuddy://

**Comment by pietbrauer** [comment]
You could also leave it empty on `dev.xing.com` and check `I don't want to use a callback domain.`. This will enable your callback URL as `xingapp,your_consumer>` [as implemented](https://github.com/xing/XNGAPIClient/blob/master/XNGAPIClient/XNGAPIClient.m#L90).

**Comment by Sam0711er** [comment]
Hey Piet, Finally the API Team replied to me and set the check. Still doesn't work. What to do now? I need a response pretty fast!

**Comment by pietbrauer** [comment]
What does the error say now? In general it works for other apps using this library, so it has to be a configuration issue. Building up pressure doesn't help here.

**Comment by Sam0711er** [comment]
Hey, thanks for responding so fast!: Here's the error:

CALLBACK_URL_NOT_ALLOWED
The oauth_callback URL isn't valid. The callback domain for this consumer is 'eckert.sam.url://xingapp*othercharacters*dd89' or 'eckert.sam.url://*othercharacters*dd89'

**Comment by pietbrauer** [comment]
Ok, I talked to the API team and we removed every callback URL completely from the database. Could you try again please?

**Comment by Sam0711er** [comment]
Doesn't work, let me quickly get the error code from the developer console.

**Comment by Sam0711er** [comment]
CALLBACK_URL_NOT_ALLOWED
The oauth_callback URL isn't valid. You have to use 'oob'

**Comment by pietbrauer** [comment]
Seems the callback URL is not set on the client. Which Login method are you using?

1. [`-loginOAuthWithSuccess:failure:`]
(https://github.com/xing/XNGAPIClient/blob/master/XNGAPIClient/XNGAPIClient.h#L46-L47)
2. [`loginOAuthAuthorize:loggedIn:failuire:`](https://github.com/xing/XNGAPIClient/blob/master/XNGAPIClient/XNGAPIClient.h#L60-L62)

**Comment by Sam0711er** [comment]
We're using loginOAuthAuthorize in combination with Swift

**Comment by pietbrauer** [comment]
But then you have to build your own ViewController with a WebView. You don't have to use the callback at all. See [docs](http://cocoadocs.org/docsets/XNGAPIClient/1.1.1/Classes/XNGAPIClient.html#//api/name/loginOAuthAuthorize:loggedIn:failuire:)

**Comment by Sam0711er** [comment]
But how should we create a UIWebView with that?
